src/backends/sentry_crashpad_backend.cpp

```diff
@@ -0,0 +1,199 @@
+
+    base::FilePath database(options->database_path->path);
+    base::FilePath handler(options->handler_path->path);
+
+    std::map<std::string, std::string> annotations;
+    std::map<std::string, base::FilePath> file_attachments;
+
+    // register attachments
+    for (sentry_attachment_t *attachment = options->attachments; attachment;
+         attachment = attachment->next) {
+        file_attachments.emplace(
+            attachment->name, base::FilePath(attachment->path->path));
+    }
+
+    // and add the serialized event, and two rotating breadcrumb files
+    // as attachments and make sure the files exist
+    data->event_path
+        = sentry__path_join_str(current_run_folder, "__sentry-event");
+    data->breadcrumb1_path
+        = sentry__path_join_str(current_run_folder, "__sentry-breadcrumb1");
+    data->breadcrumb2_path
+        = sentry__path_join_str(current_run_folder, "__sentry-breadcrumb2");
+
+    sentry__path_touch(data->event_path);
+    sentry__path_touch(data->breadcrumb1_path);
+    sentry__path_touch(data->breadcrumb2_path);
+
+    file_attachments.emplace(
+        "__sentry-event", base::FilePath(data->event_path->path));
+    file_attachments.emplace(
+        "__sentry-breadcrumb1", base::FilePath(data->breadcrumb1_path->path));
+    file_attachments.emplace(
+        "__sentry-breadcrumb2", base::FilePath(data->breadcrumb2_path->path));
+
+    std::vector<std::string> arguments;
+    arguments.push_back("--no-rate-limit");
+
+#ifdef _WIN32
+    // Temporary fix for Windows
+    arguments.push_back("--no-upload-gzip");
+#endif
+
+    // initialize database first and check for user consent.  This is going
+    // to change the setting persisted into the crashpad database.  The
+    // update to the consent change is then reflected when the handler starts.
+    g_db = crashpad::CrashReportDatabase::Initialize(database);
+    user_consent_changed(backend);
+
+    crashpad::CrashpadClient client;
+    std::string url(sentry__dsn_get_minidump_url(&options->dsn));
+    bool success = client.StartHandlerWithAttachments(handler, database,
+        database, url, annotations, file_attachments, arguments,
+        /* restartable */ true,
+        /* asynchronous_start */ false);
+
+    if (success) {
+        SENTRY_DEBUG("started crashpad client handler.");
+    } else {
+        SENTRY_DEBUG("failed to start crashpad client handler.");
+        return;
+    }
+
+    if (!options->system_crash_reporter_enabled) {
+        // Disable the system crash reporter. Especially on macOS, it takes
+        // substantial time *after* crashpad has done its job.
+        crashpad::CrashpadInfo *crashpad_info
+            = crashpad::CrashpadInfo::GetCrashpadInfo();
+        crashpad_info->set_system_crash_reporter_forwarding(
+            crashpad::TriState::kDisabled);
+    }
+}
+
+static void
+flush_scope(sentry_backend_t *backend, const sentry_scope_t *scope)
+{
+    sentry_value_t event = sentry_value_new_object();
```

Comment:
_nit_: Comment here that we deliberately do **not** create an event so that the scope payload does not contain default values like the `timestamp` or `event_id`.